nestkernel/connection_manager.h

```diff
@@ -582,6 +593,9 @@ class ConnectionManager : public ManagerInterface
   //! simulate.
   PerThreadBoolIndicator have_connections_changed_;
 
+  //! True if GetConnections has been called.
+  PerThreadBoolIndicator has_get_connections_been_called_;
```

Comment:
Do we strictly need this to be a `PerThreadBoolIndicator`? Wouldn't it be sufficient to have a simple `bool` (initialized `false`) and set `true` just before returning the `result` in [`ConnectionManager::get_connections()`](https://github.com/nest/nest-simulator/blob/master/nestkernel/connection_manager.cpp#L834) instead of doing this in the parallel context? That way, we'd still have less parallel operations and the flag still unset even if any of the exceptions later on in the function would trigger.

Reply:
I agree, I've converted it to a `bool` now.